addon/mixins/data-adapter-mixin.js

```diff
@@ -56,6 +56,23 @@ export default Ember.Mixin.create({
   authorizer: null,
 
   /**
+    Return parameters for the
+    {{#crossLink "SessionService/authorize:method"}}{{/crossLink}} method.
+
+    This function should return an array or "extra" parameters for the
+    `authorize()` method. By default an empty array is returned. If your
+    authorizer needs more information about the issued HTTP request, you should
+    override this method in your authorizer implementation and return any
+    data from the `xhr` and `ajaxSetup` objects.
+
+    @param {Object} xhr the jQuery xhr object.
+    @param {Object} ajaxSetup the jQuery ajaxSetup object.
+   */
+  getAuthorizerParams(/*xhr, ajaxSetup*/) {
+    return [];
+  },
```

Comment:
Not so happy with adding an extra method for this actually. An alternative would be to always pass e.g. the XHR but that would mean that the `DataAdapterMixin` would only be compatible with authorizers that take the XHR as an argument.

Actually the goal should be that the `DataAdapterMixin` works with authorizers that take additional arguments out of the box.